hts.c

```diff
@@ -70,7 +75,10 @@ int hts_verbose = HTS_LOG_WARNING;
 
 const char *hts_version()
 {
-    return HTS_VERSION_TEXT;
+    static char version[256];
+    snprintf(version, sizeof(version), "%s (with htscodecs %s)",
+             HTS_VERSION_TEXT, htscodecs_version());
+    return version;
```

Comment:
`hts_version()` is documented as returning literally `N.N[.N]` or `N.N[.N]-N-gCOMMITHASH[-dirty]` and e.g. this is used to construct a one-word string in a fixed format in `samtools --version-only`.

I'd suggest leaving `hts_version()` as is and adding a function (e.g. `hts_htscodecs_version()`) declared in _htslib/hts.h_ alongside `hts_version()` that just forwards the result of htscodecs's function. Then samtools could use that explicitly to produce either human-readable or machine-readable information e.g.

```
$ samtools --version
samtools 1.11-30-g63f345d
Using htslib 1.11-104-ga117153-dirty
Using htscodecs 1.0
Copyright (C) 2021 Genome Research Ltd.
[…]

$ samtools --version-only
1.11-30-g63f345d+htslib-1.11-104-ga117153-dirty+htscodecs-1.0
```

(See also [GNU's note about the Using lines](https://www.gnu.org/prep/standards/html_node/_002d_002dversion.html); I think putting the word `Using` was my small addition to distinguish these from the main version number.)

Reply:
Hmm, yes it does break `samtools --version-only` a bit.  I was hoping I could get away without adding another function.  Another alternative would be to add it into the `hts_feature_string()` output, although I'm not sure if it really fits well there either.

Reply:
> … add it into the `hts_feature_string()` output …

And you could get the raw codecs version number — if you really wanted to — with something like `hts_test_feature(HTS_FEATURE_HTSCODECS)`. I like it!

Reply:
Yes, that's a good plan.  I'll make it so.

Reply:
I agree with James that once this settles down, it mostly won't change much. So recording it in the `hts_feature_string()` that Samtools is already printing out displays it as useful background info, and avoids needing extra code to call it out in the version number part.

It's win/win and a vindication of the introspection stuff! One thing I hesitate to mention… It does feel like `hts_test_feature()` is a wee bit misnamed… any appetite for simplifying that to `hts_feature(unsigned int id)`?

Reply:
It isn't ideally named, but we already have `hts_features()`, so `hts_feature()` is maybe a bit too similar?

Reply:
Gimme all `hts_features()` vs gimme one `hts_feature(HTS_FEATURE_THIS_ONE)` I think is okay :shrug:

I mostly want to fend off people saying “is `hts_test_feature()` just about hslib's testing functionality” but that could be done via documentation comments if it ever became an issue…

Reply:
I've added a commit which moves the htscodecs version to the introspection API.  It currently has no opinion on what `hts_test_feature()` is called, as any change to that really ought to be done separately.

Reply:
> It does feel like `hts_test_feature()` is a wee bit misnamed… any appetite for simplifying that to `hts_feature(unsigned int id)`?

My thought process is if you're calling it on a specific feature then you're most likely testing whether that feature exists, otherwise you're just asking for the whole list of them.  However I'm ambivalent and could live with hts_feature too if others prefer it.